### PR TITLE
297 edit config form

### PR DESCRIPTION
**Pull request by Elkrival**
This pr adds the edit config form page reusing the form components and adding a global notification state. It adds the functionality for the select component value to not be deleted.

### CODE REVIEW

File: server/controllers/configurationsController/configurationsController.test.js [hunk 1]
```diff
@@ -176,4 +176,41 @@ describe('ConfigurationsController', () => {
...
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          data: config,
+        })
+      })
+    })
+
+    describe('When unsuccessful', () => {
+      it('sends a status of 400 with a message when database there is an error', async () => {
```
Comment (taylorkearns) [inline server/controllers/configurationsController/configurationsController.test.js:199]:
```suggestion
      it('sends a status of 400 with a message when there is an error', async () => {
```

**Review by taylorkearns — APPROVED** [review]

File: server/routes/configurations.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ router
 
 router
   .route(v1Routes.config.show)
+  .get(ensureUser, ConfigurationsController.findOne)
```
Comment (mikestone14) [inline server/routes/configurations.js:16]: this should be the `show` action.

File: views/forms/ConfigAssessmentFormFields/index.jsx [hunk 3]
```diff
@@ -9,12 +9,14 @@ const ConfigAssessmentFormFields = ({
...
   index,
-  width,
+  id,
   onCopy,
   onRemove,
+  width,
 }) => {
   return (
     <ConfigurationCategoryCard
+      key={id + 'card'}
```
Comment (mikestone14) [inline views/forms/ConfigAssessmentFormFields/index.jsx:19]: you're not rendering an array so you shouldn't need these key props.

File: views/forms/ConfigFields/index.jsx [hunk 4]
```diff
@@ -30,7 +30,8 @@ const ConfigFormFields = ({
               colors={colors}
               control={control}
               index={index}
-              key={field.id}
+              key={id}
+              id={id}
```
Comment (mikestone14) [inline views/forms/ConfigFields/index.jsx:34]: you have `const { id, ...rest } = field` above and `{...rest}` below. If you change `{...rest}` to `{...field}` you can remove the destructuring above and keep the `key` as `key={field.id}`.

File: views/models/UserConfigModel/UserConfigModel.test.js [hunk 5]
```diff
@@ -55,4 +56,54 @@ describe('Models - User Confg', () => {
       })
     })
   })
+
+  describe(UserConfigModel.processConfigToFormFields, () => {
+    it('returns a current configuration into form values', async () => {
+      const currentConfiguration = {
+        config: {
+          0: [
```
Comment (mikestone14) [inline views/models/UserConfigModel/UserConfigModel.test.js:64]: what do the keys in the config object represent (what is `0` here)?

**Pull request merged**